src/pymovements/gaze/gaze_dataframe.py

```diff
@@ -296,15 +296,22 @@ def split(self, by: Sequence[str]) -> list[GazeDataFrame]:
             A list of new GazeDataFrame instances, each containing a partition of the
             original data with all metadata and configurations preserved.
         """
+        frames = self.frame.partition_by(by=by)
+        # we cannot directly zip because we need to check if by is in events columns
+        events_list = self.events.split(by) if by in self.events.columns else [
```

Comment:
this won't work if `by` is a list of strings. please add a test case to make sure list of strings are supported for `by`.

Reply:
done